### PR TITLE
Recalculate Tooltip's top/bottom position if initial position doesn't fit

### PR DESCRIPTION
**Pull request by chriscain**
Small optimization and usually an edge case, but in the case that the desired top/bottom position can’t fit the tooltip, there should be a recalculation so that the tooltip is not covering the desired target. This _still_ doesn’t account for everything (i.e. an element in the corner) but this is a step in the right direction.

### CODE REVIEW

**Comment by gilbarbara** [comment]
"This calculation was assuming the body wasn’t already scrolled, which was stupid."
Thanks

**Comment by gilbarbara** [comment]
Just messing around. 😀

**Comment by IanVS** [comment]
Looks like Travis hung, maybe you can restart?

**Comment by IanVS** [comment]
Rebased on master.  Please let us know if you have any feedback.

**Comment by gilbarbara** [comment]
I think I fixed this in 2c932daa16229e521e12f5e37a1d5c1f8c1fab54

File: src/scripts/Joyride.jsx [hunk 1]
```diff
@@ -883,10 +883,17 @@ class Joyride extends React.Component {
...
+    else if (/^right/.test(position) && (rect.left + rect.width + (component.width + tooltipOffset)) > body.getBoundingClientRect().width) {
       position = 'bottom';
     }
 
+    if (/^top/.test(position) && (rect.top + body.scrollTop) - (component.height + tooltipOffset) < 0) {
+      position = 'bottom';
+    }
+    else if (/^bottom/.test(position) && (rect.bottom + body.scrollTop) + (component.height + tooltipOffset) > getDocHeight()) {
+      position = 'top';
+    }
```
Comment (IanVS) [inline src/scripts/Joyride.jsx]: @gilbarbara I think this part of the PR is not handled so far by your commit, is it?

Comment (gilbarbara) [inline src/scripts/Joyride.jsx]: Nope. Let's add it and I'll try it in the demo afterwards.

**Pull request merged**